pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -15,24 +15,124 @@
+            print(np.allclose(contraction_matrix, np.eye(tensor.shape[0])))
+
+    .. code-block:: pycon
+
+        True
+        True
+    """
+
+    is_right_canonical = True
+    d_shapes = []
+    for tensor in mps[1:-1]:
+        # Right-canonical definition
+        d_shapes += tensor.shape
+        input_matrix = qml.math.tensordot(tensor, tensor.conj(), axes=([1, 2], [1, 2]))
+        if not qml.math.allclose(input_matrix, qml.math.eye(tensor.shape[0])):
+            is_right_canonical = False
+
+    if is_right_canonical:
+        return mps
+
+    max_bond_dim = qml.math.max(d_shapes)
+
+    n_sites = len(mps)
+    output_mps = [None] * n_sites
+
+    # Procedure analogous to the left-canonical conversion but starting from the right and storing the Vd
+    for i in range(n_sites - 1, 0, -1):
+        chi_left, d, chi_right = mps[i].shape
+        input_matrix = mps[i].reshape(chi_left, d * chi_right)
+
+        u_matrix, s_diag, vd_matrix = qml.math.linalg.svd(input_matrix, full_matrices=False)
+
+        # Truncate SVD components if needed
+        chi_new = min(int(max_bond_dim), len(s_diag))
+        u_matrix = u_matrix[:, :chi_new]
+        s_diag = s_diag[:chi_new]
+        vd_matrix = vd_matrix[:chi_new, :]
+
+        # Store Vd reshaped as an MPS tensor in the output MPS
+        output_mps[i] = vd_matrix.reshape(chi_new, d, chi_right)
+
+        # Contract U with diag(S) and merge it with the preceding MPS tensor, preserving the canonical structure
+        mps[i - 1] = qml.math.tensordot(
+            mps[i - 1], u_matrix @ qml.math.diag(s_diag), axes=([2], [0])
+        )
+
+    output_mps[0] = mps[0]
+    return output_mps
+
+
 class MPSPrep(Operation):
     r"""Prepares an initial state from a matrix product state (MPS) representation.
 
     .. note::
 
-        Currently, this operator can only be used with ``qml.device(“lightning.tensor”)``.
-
+        This operator is natively supported on the ``lightning.tensor`` device, which is designed to run MPS
+        structures efficiently. For other devices, this operation prepares the state vector represented by the
+        MPS using a gate-based decomposition from Eq. (23) in `[arXiv:2310.18410]
+        <https://arxiv.org/pdf/2310.18410>`_, which requires the right canonicalization of the MPS via
+        ``right_canonicalize`` and auxiliary qubits to be given via ``work_wires``.
 
     Args:
-        mps (List[Array]):  list of arrays of rank-3 and rank-2 tensors representing an MPS state as a
-            product of site matrices. See the usage details section for more information.
+        mps (list[TensorLike]):  list of arrays of rank-3 and rank-2 tensors representing an MPS state
+            as a product of site matrices. See the usage details section for more information.
 
         wires (Sequence[int]): wires that the template acts on
+        work_wires (Sequence[int]): list of extra qubits needed in the decomposition. If the maximum bond dimension
```

Comment:
What happens if the user puts a wrong work_wires? Can we just compute the acceptable k and verify of the work_wire matches that? If not, tell users that work_wires should be 1, 2, ...